test/dt_arith.c

```diff
@@ -2694,18 +2694,18 @@ my_isnan(dtype_t type, void *val)
     if (FLT_FLOAT == type) {
         float x = 0.0;
         HDmemcpy(&x, val, sizeof(float));
-        retval = (x != x);
+        retval = !H5_FLT_ABS_EQUAL(x, x);
```

Comment:
Don't compare floats, doubles, long doubles using != here. Use library macro to compare with an epsilon value instead.